examples/platforms/simulation/radio.c

```diff
@@ -221,6 +222,54 @@ static uint16_t crc16_citt(uint16_t aFcs, uint8_t aByte)
     return (aFcs >> 8) ^ sFcsTable[(aFcs ^ aByte) & 0xff];
 }
 
+
+#if OPENTHREAD_SIMULATION_VIRTUAL_TIME == 1
+const char * radioStateToString(otRadioState aState)
+{
+    char const * ans;
+    switch (aState)
+    {
+    case OT_RADIO_STATE_RECEIVE:
+        ans = "rx";
+        break;
+    case OT_RADIO_STATE_TRANSMIT:
+        ans = "tx";
+        break;
+    case OT_RADIO_STATE_DISABLED:
+        ans = "off";
+        break;
+    case OT_RADIO_STATE_SLEEP:
+        ans = "sleep";
+        break;
+    default:
+        assert(false);
+    }
+    return ans;
+}
+
+void reportRadioStatusToOtns(void)
+{
+    struct Event event;
+    
+    event.mDelay      = 0;
+    event.mEvent      = OT_SIM_EVENT_OTNS_STATUS_PUSH;
+
+    int n = snprintf((char *)event.mData, sizeof(event.mData), "radio_state=%s", radioStateToString(sState));
+    assert(n > 0);
+    event.mDataLength = n;
+
+    otSimSendEvent(&event);
+}
+#endif
+
+void setRadioState(otRadioState aState)
+{
+    sState = aState;
```

Comment:
I think it's possible that `sState` equals to `aState`. Should we skip `reportRadioStatusToOtns` in such cases?

If such cases happen a lot, which I am not sure, it would cause significant overhead to OTNS simulation.

Reply:
Nice! Good point.